src/bytes.rs

```diff
@@ -16,7 +16,7 @@ pub enum Encoding {
 #[derive(PartialEq)]
 pub enum Confidence {
     /// This encoding might be the one we need
-    Tentative,
+    Tentative(f32),
```

Comment:
probably you could save some memory here with a u8, but it is only marginal, so feel free to ignore